File: leetcode/april2024/c++/09_time_needed_to_buy_tickets.cpp
// CODE

class Solution {
    public:
    int timeRequiredToBuy(vector<int>& tickets, int k) {
        if (tickets[k] == 1) return k + 1;
        int n = tickets.size(), res = 0;
        while (tickets[k] > 0) {
            for (int i = 0; i < n; i++) {
                if (tickets[i] != 0) {
                    tickets[i]--;
                    res++;
                }
                if (tickets[k] == 0) return res;
            }
        }
        return res;
    }
};

File: leetcode/april2024/python/09_time_needed_to_buy_tickets.py
# CODE

class Solution:
    def timeRequiredToBuy(self, tickets: List[int], k: int) -> int:
        if tickets[k] == 1: return k + 1
        n = len(tickets)
        res = 0
        while tickets[k] > 0:
            for i in range(n):
                if tickets[i] != 0:
                    tickets[i] -= 1
                    res += 1
                if tickets[k] == 0: return res
        return res